### PR TITLE
Fix error handling

### LINKED ISSUE #213: [PE-D][Tester C] Large and big number for delete function
Works for small numbers 

But when numbers get really large, the command doesnt work as expected. As below: it gives an invalid command format error. 

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: junlee1991/ped#5

### LINKED ISSUE #227: [PE-D][Tester C] Large numbers
Sp command fails when very large numbers are input

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: junlee1991/ped#7

### PR DESCRIPTION
**Pull request by vigonometry**
- Fix throwing of correct exceptions for edit and delete features: Closes #190 
- Fix handling of numbers larger than `Integer.MAX_VALUE`: Closes #227 and Closes #213

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -31,13 +31,21 @@ public class ParserUtil {
...
+    private static final String MESSAGE_OUT_OF_RANGE = "Index provided is greater than maximum number of records.";
     /**
      * Parses {@code oneBasedIndex} into an {@code Index} and returns it. Leading and trailing whitespaces will be
      * trimmed.
      * @throws ParseException if the specified index is invalid (not non-zero unsigned integer).
      */
     public static Index parseIndex(String oneBasedIndex) throws ParseException {
         String trimmedIndex = oneBasedIndex.trim();
+        System.out.println(trimmedIndex.compareTo(MAX_SIZE));
+        if (trimmedIndex.compareTo(MAX_SIZE) >= 0) {
```
Comment (praveenkrishna0512) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:46]: Does the string comparison with the trimmed index work as intended?

**Review by praveenkrishna0512 — COMMENTED** [review]
LGTM but the test cases are failing due to checkstyle

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (vigonometry) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Updated to include length comparison in addition to lexicographic to fix erroneous exceptions thrown

**Review by praveenkrishna0512 — APPROVED** [review]
LGTM!

**Pull request merged**